contractcourt/anchor_resolver.go

```diff
@@ -161,14 +169,17 @@ func (c *anchorResolver) Resolve() (ContractResolver, error) {
 
 	// Update report to reflect that funds are no longer in limbo.
 	c.reportLock.Lock()
-	if anchorRecovered {
+	if outcome == channeldb.ResolverOutcomeClaimed {
 		c.currentReport.RecoveredBalance = c.currentReport.LimboBalance
 	}
 	c.currentReport.LimboBalance = 0
+	report := c.currentReport.resolverReport(
+		spendTx, channeldb.ResolverTypeAnchor, outcome,
+	)
 	c.reportLock.Unlock()
 
 	c.resolved = true
-	return nil, nil
+	return nil, c.PutResolverReport(nil, report)
```

Comment:
I still don't fully understand why `PutResolverReport` needs to be called directly. I'd think that always calling `CheckPoint` in resolvers makes it clearer what is going on. It checkpoints the full state (internal + reports) of a resolver and that is the only function that is used.

I know there is access via the arb config to other functions, but maybe that isn't a good thing anyway.